### ISSUE TITLE
Optimized model seems slower than original

### ISSUE THREAD

**Issue by BitCircuit** [Azure/MS-AMP#172]
I have tested examples/mnist.py in a docker container on a PC with RTX 4090. 
Here is what I found:
| MNIST | Accuracy | .pt file size | VRAM occupied (may not be accurate) | avg. inference time |
|-------|--------|---------|---------|-------------|
| Original | 98.75 | 4.6 MB | 492 MB | 0.4 ms |
| Optimized by MS-AMP | 98.76 | 2.4 MB | 512 MB | 1.2 ms |

I also found that each layer in original model has data type "torch.float32", and mixed "torch.float32" and "torch.float16" in optimized model. I guess that's because pytorch has not supported float8 yet. 

However, any idea why the optimized model takes longer time for inference?

**Comment by wkcn** [comment]
Hi @BitCircuit , thanks for your attention to our work!

In MS-AMP, the additional operations for FP8 quantization and de-quantization take up part of time.

When training a small model, the speedup of FP8 calculation cannot make up for the cost caused by quantization.

In my experience, FP8 takes less time when training a model whose parameter size is larger than 1B.

**Comment by BitCircuit** [comment]
Thank you for your reply. I thought MS-AMP was able to train the model as in float8 weight format directly. From your response, I assumed that MS-AMP trains the model as QAT (with QDQ nodes inserted)?

> In my experience, FP8 takes less time when training a model whose parameter size is larger than 1B.

After second thought, I realized that MS-AMP is a tool to optimize training speed instead of inferencing speed. Is that right?

**Comment by wkcn** [comment]
@BitCircuit Yes. MS-AMP is an FP8 mixed-precision training framework, optimizing the memory usage and training speed.
